### PR TITLE
Cp flags as a struct

### PR DESCRIPTION
**Pull request by ananion-cation**
Previous problems with cp in bb, so I fixed the flags.

### CODE REVIEW

File: cmds/cp/cp.go [hunk 1]
```diff
@@ -34,33 +34,32 @@ import (
...
 	input     = bufio.NewReader(os.Stdin)
 	// offchan is a channel used for indicate the nextbuffer to read with worker()
 	offchan = make(chan int64, 0)
 	// zerochan is a channel used for indicate the start of a new read file
 	zerochan = make(chan int, 0)
 )
 
+
+
+type cpFlags struct{
```
Comment (hugelgupf) [inline cmds/cp/cp.go]: just make it type options or type flags. "cp.cpFlags" stutters

File: cmds/cp/cp_test.go [hunk 2]
```diff
@@ -33,13 +33,13 @@ const (
...
 // resetFlags is used to reset the cp flags to default
-func resetFlags() {
-	nwork = 1
-	recursive = false
-	ask = false
-	force = false
-	verbose = false
-	symlink = false
+func resetFlags(flags cpFlags) {
+	flags.nwork = 1
```
Comment (hugelgupf) [inline cmds/cp/cp_test.go]: default is runtime.NumCPU, no?

File: cmds/cp/cp_test.go [hunk 3]
```diff
@@ -187,10 +187,11 @@ func isEqualTree(src, dst string) (bool, error) {
 // TestCpsSimple make a simple test for copy file-to-file
 // cmd-line equivalent: cp file file-copy
 func TestCpSimple(t *testing.T) {
+	var flags cpFlags
 	tempDir, err := ioutil.TempDir(testPath, "TestCpSimple")
-	if remove {
+	/*if remove {
```
Comment (hugelgupf) [inline cmds/cp/cp_test.go]: why comment out?

File: cmds/cp/cp_test.go [hunk 4]
```diff
@@ -202,9 +203,10 @@ func TestCpSimple(t *testing.T) {
 	dstFname := fmt.Sprintf("cpfile_%v_dst_copied", indentifier)
 	dstFpath := filepath.Join(tempDir, dstFname)
 
-	if err := copyFile(srcFpath, dstFpath, false); err != nil {
+	if err := copyFile(srcFpath, dstFpath, false, flags); err != nil {
 		t.Fatalf("copyFile %v -> %v failed: %v", srcFpath, dstFpath, err)
 	}
+/*
```
Comment (hugelgupf) [inline cmds/cp/cp_test.go]: why comment out?

File: cmds/cp/cp_test.go (hunk 2 above)
Comment (ananion-cation) [reply cmds/cp/cp_test.go]: removed this parameter

**Comment by rminnich** [comment]
This failed due to a bug in the Go compiler, we'll have to work it out this week. Sheesh.

To double check which one
cd bb
GOARCH=ppc64le ./bb
and
GOARCH=arm64 ./bb

see which one fails.

**Comment by rminnich** [comment]
THis builds for me but you need to rebase. Don't forget to always rebase before you push a commit.

**Comment by hugelgupf** [comment]
Going to close this for now. Reopen it if you do work on it again :)

**Pull request closed without merging**